source/MRViewer/MRObjectHighlighter.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "MRMesh/MRMeshFwd.h"
+#include "MRViewer/exports.h"
+
+namespace MR
+{
+
+// This is used to temporarily change object colors (and other visual properties) in the scene.
+// NOTE: Currently at most one highlighter can operate at a time.
+// The workflow is as follows:
+// 1. Create `ObjectHighlighter` as a class member in your plugin.
+// 2. Using the constructor parameters, you can customize which fields we roll back. The default behavior should usually be good enough.
+// 3. Call `updateColors()` every frame in `preDraw_()` to set the desired colors.
+// 4. Call `restoreOriginalState()` in `onDisable_()`.
+class ObjectHighlighter
```

Comment:
can it be applied to colormaps etc? mb rename?

Reply:
Yes, it can work with anything, if you pass the appropriate rollback rules to the constructor. I think "highlighter" sounds ok even for colormaps, but I'm open to suggestions for a better name.